source/NVDAObjects/IAccessible/adobeAcrobat.py

```diff
@@ -119,61 +119,96 @@ def _isEqual(self, other):
 			return self.accID == other.accID
 		return super(AcrobatNode, self)._isEqual(other)
 
-	def _getNodeMathMl(self, node):
+	@staticmethod
+	def getMathMLAttributes(node, attrList: list) -> str:
+		"""Get the MathML attributes in 'attrList' for a 'node' (MathML element)."""
+		attrValues = ""
+		for attr in attrList:
+			# "NSO" comes from the PDF spec
+			val = node.GetAttribute(attr, "NSO")
+			if val:
+				attrValues += f' {attr}="{val}"'
+		return attrValues
+
+	def _getNodeMathMl(self, node) -> str:
+		"""Traverse the MathML tree and return an XML string representing the math"""
+
 		tag = node.GetTagName()
-		yield "<%s" % tag
-		# Output relevant attributes.
-		if tag == "mfenced":
-			for attr in "open", "close", "separators":
-				val = node.GetAttribute(attr, "XML-1.00")
-				if val:
-					yield ' %s="%s"' % (attr, val)
-		yield ">"
+		answer = f"<{tag}"
+		# Output relevant attributes
+		id = node.GetID()
+		if id:
+			answer += f' id="{id}"'
+		# The PDF interface lacks a way to get all the attributes, so we have to get specific ones
+		# The attributes below affect accessibility
+		answer += AcrobatNode.getMathMLAttributes(node, ["intent", "arg"])
+		match tag:
+			case "mi" | "mn" | "mo" | "mtext":
+				answer += AcrobatNode.getMathMLAttributes(node, ["mathvariant"])
+			case "mfenced":
+				answer += AcrobatNode.getMathMLAttributes(node, ["open", "close", "separators"])
+			case "menclose":
+				answer += AcrobatNode.getMathMLAttributes(node, ["notation", "notationtype"])
+			case "annotation-xml" | "annotation":
+				answer += AcrobatNode.getMathMLAttributes(node, ["encoding"])
+			case "ms":
+				answer += AcrobatNode.getMathMLAttributes(node, ["open", "close"])
+			case _:
+				pass
+		answer += ">"
 		val = node.GetValue()
 		if val:
-			yield val
+			answer += val
 		else:
 			for childNum in range(node.GetChildCount()):
 				try:
 					subNode = node.GetChild(childNum).QueryInterface(IPDDomElement)
 				except COMError:
 					continue
 				for sub in self._getNodeMathMl(subNode):
-					yield sub
-		yield "</%s>" % tag
+					answer += sub
+		return answer + f"</{tag}>"
 
 	def _get_mathMl(self) -> str:
```

Comment:
this seems like it can return None, should it return an empty string at the end or should the type hint be updated?

Reply:
if it doesn't raise an exception, `_get_mathML` always returns a `str` doesn't it (unless I am misreading the code the three possible `return` are all strings?

Reply:
Nevermind - yes I misread this